pallet/message-transact/src/lib.rs

```diff
@@ -17,6 +17,7 @@
 // along with Darwinia. If not, see <https://www.gnu.org/licenses/>.
 
 #![cfg_attr(not(feature = "std"), no_std)]
+#![cfg_attr(not(test), deny(unused_crate_dependencies))]
```

Comment:
> This lint is "allow" by default because it can provide false positives depending on how the build system is configured. For example, when using Cargo, a "package" consists of multiple crates (such as a library and a binary), but the dependencies are defined for the package as a whole. If there is a dependency that is only used in the binary, but not the library, then the lint will be incorrectly issued in the library.

I guess that regards tests is treated as another binary. The compiler refuses to pass unless you move files under the `tests` folder. But this seems less clear than the current layout.